tests/End2EndTestCase.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/**
+ * This file is part of the contentful.php package.
+ *
+ * @copyright 2015-2017 Contentful GmbH
+ * @license   MIT
+ */
+
+namespace Contentful\Tests\Delivery;
+
+use Contentful\Delivery\Client;
+use Contentful\Log\ArrayLogger;
+use Symfony\Component\Filesystem\Filesystem;
+
+class End2EndTestCase extends \PHPUnit_Framework_TestCase
+{
+    const CACHE_DIR = __DIR__ . '/../build/cache';
+
+    /**
+     * @var Client[]
+     */
+    private static $clients = [];
+
+    public static function setUpBeforeClass()
+    {
+        self::$clients = [
+            'cfexampleapi' => new Client('b4c0n73n7fu1', 'cfexampleapi'),
+            'cfexampleapi_preview' => new Client('e5e8d4c5c122cf28fc1af3ff77d28bef78a3952957f15067bbc29f2f0dde0b50', 'cfexampleapi', true),
+            'cfexampleapi_cache' => new Client('b4c0n73n7fu1', 'cfexampleapi', false, null, ['cacheDir' => self::CACHE_DIR]),
+            'cfexampleapi_logger' => new Client('b4c0n73n7fu1', 'cfexampleapi', false, null, ['logger' => new ArrayLogger()]),
+            'cfexampleapi_tlh' => new Client('b4c0n73n7fu1', 'cfexampleapi', false, 'tlh'),
+            'cfexampleapi_invalid' => new Client('e5e8d4c5c122cf28fc1af3ff77d28bef78a3952957f15067bbc29f2f0dde0b50', 'cfexampleapi'),
+            '88dyiqcr7go8' => new Client('668efbfd9e398181166dec5df5a500aded96dbca2916646a3c7ec37082a7b756', '88dyiqcr7go8'),
+            '88dyiqcr7go8_preview' => new Client('81c469d7241ca02349388602dfc14107157063a6901c378a56e1835d688970bf', '88dyiqcr7go8', true),
+            'bc32cj3kyfet_preview' => new Client('8740056d546471e0640d189615470cc12ce2d3188332352ecfb53edac59c4963', 'bc32cj3kyfet', true),
+        ];
+    }
+
+    public function setUp()
```

Comment:
Yeah, I was trying to make something smarter than this, but in the end the performance impact is basically negligible...

Maybe I could implement this using an annotation like I did with `@requires`, something like `@uses-cache`?